File: ios/modules/IBeacon.swift
import Foundation

struct IBeacon: Beacon {
    let type: String = "iBeacon"
    let uuid: String
    let major: Int
    let minor: Int
    let rssi: Int

    func toMap() -> [String: Any?] {
        return [
            "type": type,
            "uuid": uuid,
            "major": major,
            "minor": minor,
            "rssi": rssi,
        ]
    }
}

File: android/src/main/java/expo/modules/beacons/modules/IBeacon.kt
package expo.modules.beacons.modules

data class IBeacon(
    override val uuid: String,
    val address: String,
    val name: String?,
    val major: Int,
    val minor: Int,
    val rssi: Int,
    val txPower: Int,
    val adHex: String,
    val battery: Int?
): Beacon {

    override fun toMap(): Map<String, Any?> = mapOf(
        "type" to "iBeacon",
        "address" to address,
        "name" to name,
        "uuid" to uuid,
        "major" to major,
        "minor" to minor,
        "rssi" to rssi,
        "txPower" to txPower,
        "adHex" to adHex,
        "battery" to battery
    )
}